### PR TITLE
Fixes misuse of transform in some animate() calls

### PR DESCRIPTION
**Pull request by The-Moon-Itself**
## About The Pull Request
This pull request reworks some animate calls to not hardset the transform of mobs to the identity matrix, potentially causing bugs with other objects that affect the transform such as gigantism/dwarfsim genes, growth serum, being buckled to a bed, etc.

The 3 things that were reworked are the waddle component, pituitary disruption virus symptom (which used waddling previously), and the dance machine.

## Why It's Good For The Game
While messing with transforms in weird ways can be fun, rotating your character or getting extremely big or small, the potential for this bug to be game breaking in combat is too great, especially for becoming extremely small. Size changes should be balanced.

## Testing Photographs and Procedure

<details>

<summary>Pituitary Disruption and Waddling</summary>

https://user-images.githubusercontent.com/51838176/155658299-4719b6ce-d194-43e4-80ca-f52a251cd654.mp4

</details>

<details>
<summary>Dance Machine</summary>

https://user-images.githubusercontent.com/51838176/155658711-59e0c8c5-ba40-4e3a-bf57-c8f80ae71b32.mp4

(Music is only for testing and is not included in the PR)

</details>

## Changelog
:cl:
fix: Waddling, pituitary disruption, and the dance machine now properly reset transforms
/:cl:

### CODE REVIEW

File: code/datums/components/waddling.dm [hunk 1]
```diff
@@ -21,6 +21,8 @@
 /datum/component/waddling/proc/Waddle()
 	SIGNAL_HANDLER
 
-	animate(parent, pixel_z = 4, time = 0)
-	animate(pixel_z = 0, transform = turn(matrix(), pick(-12, 0, 12)), time=2)
-	animate(pixel_z = 0, transform = matrix(), time = 0)
+	var/rot_degrees = pick(-12, 0, 12)
+	var/mob/living/L = parent
```
Comment (Archanial) [inline code/datums/components/waddling.dm]: This component isn't only made for living

File: code/datums/components/waddling.dm [hunk 2]
```diff
@@ -21,6 +21,8 @@
 /datum/component/waddling/proc/Waddle()
 	SIGNAL_HANDLER
 
-	animate(parent, pixel_z = 4, time = 0)
-	animate(pixel_z = 0, transform = turn(matrix(), pick(-12, 0, 12)), time=2)
-	animate(pixel_z = 0, transform = matrix(), time = 0)
+	var/rot_degrees = pick(-12, 0, 12)
```
Comment (Archanial) [inline code/datums/components/waddling.dm:24]: What was wrong with previous implementation?

File: code/datums/components/waddling.dm (hunk 1 above)
Comment (The-Moon-Itself) [reply code/datums/components/waddling.dm]: I’m not sure why I put that typecast there, I’ll get rid of it when I’m back at my computer.

File: code/datums/components/waddling.dm (hunk 2 above)
Comment (The-Moon-Itself) [reply code/datums/components/waddling.dm:24]: The previous implementation always set the transform to the identity matrix when it was done, which overrides size changes and rotations. This method undoes the rotation without incorrect assumptions about the original state.

File: code/datums/components/waddling.dm (hunk 1 above)
Comment (The-Moon-Itself) [reply code/datums/components/waddling.dm]: I changed the typecast to just atom. Anything less broad an transform won't be a member of the class anymore.

Comment (Archanial) [reply code/datums/components/waddling.dm]: This cast isn't needed

Comment (The-Moon-Itself) [reply code/datums/components/waddling.dm]: parent is a datum, which doesn't have the transform variable, it needs to be typecasted to an atom to index transform for the rotations.

Comment (Archanial) [reply code/datums/components/waddling.dm]: Then cast it to movable

Comment (The-Moon-Itself) [reply code/datums/components/waddling.dm]: It's a movable now.

**Comment by zeskorion** [comment]
could you do me a faaavor and like halve pituitary disruption's impact on size as well?

**Comment by The-Moon-Itself** [comment]
> could you do me a faaavor and like halve pituitary disruption's impact on size as well?

This PR is only supposed to be a bug fix, and I'm really trying to stop myself from scope creeping. I could make a separate pr for this if you want.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**